src/Uno.UI/UI/Xaml/VirtualizationInformation.cs

```diff
@@ -12,19 +12,59 @@ namespace Windows.UI.Xaml
 	/// </summary>
 	internal sealed class VirtualizationInformation
 	{
+		private Flags _flags;
+
 		/// <summary>
 		/// Marks this as a container generated by, eg, a <see cref="Controls.Primitives.Selector"/>, rather than an element explicitly 
 		/// defined in xaml.
 		/// </summary>
-		public bool IsGeneratedContainer { get; set; }
+		public bool IsGeneratedContainer
+		{
+			get => _flags.HasFlag(Flags.IsGeneratedContainer);
+			set => SetFlag(Flags.IsGeneratedContainer, value);
+		}
 
 		/// <summary>
 		/// Marks this as a container defined in the root of an ItemTemplate, so that it can be handled appropriately when recycled.
 		/// </summary>
-		public bool IsContainerFromTemplateRoot { get; set; }
+		public bool IsContainerFromTemplateRoot
+		{
+			get => _flags.HasFlag(Flags.IsContainerFromTemplateRoot);
+			set => SetFlag(Flags.IsContainerFromTemplateRoot, value);
+		}
+
+		/// <summary>
+		/// Marks this as an item being its own container
+		/// </summary>
+		public bool IsOwnContainer
+		{
+			get => _flags.HasFlag(Flags.IsOwnContainer);
+			set => SetFlag(Flags.IsOwnContainer, value);
+		}
 
 		public Rect Bounds { get; set; }
 
 		public Size MeasureSize { get; set; }
+
+		private void SetFlag(Flags flag, bool value)
```

Comment:
since we are optimizing down to flags already, would this method benefit of `AggressiveInlining`?

Reply:
The AOT compiler generally determines if the method should be inlined based on size, but it does not hurt to push it a bit here :)